nanoFramework.System.Net/Sockets/Socket.cs

```diff
@@ -744,12 +740,12 @@ public int Receive(byte[] buffer, int size, SocketFlags socketFlags)
         /// </remarks>
         public int Receive(byte[] buffer, int offset, int size, SocketFlags socketFlags)
         {
-            if (m_Handle == -1)
-            {
+            if (m_Handle == -1) {
                 throw new ObjectDisposedException();
             }
 
-            return NativeSocket.recv(this, buffer, offset, size, (int)socketFlags, m_recvTimeout);
+            //return NativeSocket.recv(this, buffer, offset, size, (int)socketFlags, m_recvTimeout);
+            return NativeSocket.recv(this, buffer, offset, size, (int)socketFlags, 1500);
```

Comment:
We shouldn't override the m_recvTimeout field with a fixed value, instead the requested value should be honoured.

Reply:
Sorry, this was a test case that I forgot to change back.  I'm still getting CLR_E_FAIL errors on NativeSocket.recv()
Is this something that can be fixed in the interpreter/extension?

    ++++ Exception System.Net.Sockets.SocketException - CLR_E_FAIL (5) ++++
    ++++ Message: 
    ++++ System.Net.Sockets.NativeSocket::recv [IP: 0000] ++++
    ++++ System.Net.Sockets.Socket::Receive [IP: 0019] ++++
    ++++ System.Net.Sockets.NetworkStream::Read [IP: 0062] ++++
    ++++ System.Net.InputNetworkStreamWrapper::RefillInternalBuffer [IP: 0038] ++++
    ++++ System.Net.InputNetworkStreamWrapper::Read_HTTP_Line [IP: 004b] ++++
    ++++ System.Net.HttpWebRequest::ParseHTTPResponse [IP: 002e] ++++
    ++++ System.Net.HttpWebRequest::GetResponse [IP: 0040] ++++
    ++++ ESP32server.ESP32server::processOutgoingAPIrequests [IP: 0084] ++++
    ++++ ESP32server.ESP32server+<>c::<Main>b__24_0 [IP: 0003] ++++
    ++++ Exception System.Net.WebException - 0x00000000 (5) ++++
    ++++ Message: GetResponse() failed

Reply:
All the exceptions being thrown in native code look correct. Nothing obvious is misplaced.
Can you catch the SocketException and see is the error code?

Reply:
I've modified my application so that HttpWebRequest.Timeout is not set (i.e. I commented it out).
The application no longer gets the CLR_E_FAIL exceptions from NativeSocket.recv()
I suspect the exceptions had something to do with timeouts not being properly bubbled up to the user code.
Not sure if this is something that needs to be fixed immediately, but it would be nice to document the workaround.
I.e. "Don't set HttpWebRequest.Timeout before calling GetResponse()"

Reply:
This is interesting... Follows my thoughts about this. 
1. m_timeout in HttpWebRequest is only used to setup a TimerCallback it never reaches the underlying socket.
2. It's default is WebRequest.DefaultTimeout.
3. If you're not getting an exception anymore, that's because everything happens as expected and the flow and sequence don't misbehave.
4. If you get an exception when setting the timeout, it's likely that's happening inside the TimerCallback handler code, when it does something unexpected or invalid _with_ the underlying socket. There are 2 calls to ParseHTTPResponse inside that handler, so this seems reasonable. It could be that extra checks are required there (or try catchs) to handle situations on which the expected sequence is processed gracefully _but_ as the timer callback would be running, at expiry, it will wrongly try to do things with the socket that don't make sense anymore because the response has already been received and processed.

Just a theory, I haven't done anything to prove it, just looked at the code.